### PR TITLE
BUG: copying series into empty dataframe does not preserve dataframe index name

### LINKED ISSUE #31368: Index name assignment resolution logic changes depending on whether DataFrame has any rows
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

def get_name_of_assigned_index(df_size, series_size):
  df = pd.DataFrame({}, index=pd.RangeIndex(df_size, name='df_index'))
  series = pd.Series(1.23, index=pd.RangeIndex(series_size, name='series_index'))

  df['series'] = series

  return df.index.name
```
#### Problem description

`get_name_of_assigned_index(n, m)` returns `'df_index'` *unless* `n = 0`, in which case it returns `'series_index'`.

I don't know which of these is necessarily the better behavior, but the resulting index name ought to be consistent regardless of how many - or few - rows are in the DataFrame. Subsequent operations may rely on expectations about that name, so this behavior, wherein the name changes as an unintuitive consequence of a rowless frame, could cause bugs.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Linux
OS-release: 4.3.5-smp-821.23.0.0
machine: x86_64
processor: 
byteorder: little
LC_ALL: en_US.UTF-8
LANG: None
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: None
pip: None
setuptools: unknown
Cython: None
numpy: 1.16.4
scipy: 1.2.1
pyarrow: None
xarray: None
IPython: 2.0.0
sphinx: None
patsy: 0.4.1
dateutil: 2.8.1
pytz: 2019.3
blosc: None
bottleneck: None
tables: 3.5.2
numexpr: 2.6.10dev0
feather: None
matplotlib: 3.0.3
openpyxl: None
xlrd: 1.2.0
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: 1.0.1
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.3
s3fs: None
fastparquet: None
pandas_gbq: 0+unknown
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] closes #31368
- [x] tests added / passed
   `indexing/test_partial.py:TestPartialSetting:test_index_name_empty`
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3206,9 +3206,11 @@ def _ensure_valid_index(self, value):
                     "and a value that cannot be converted to a Series"
                 ) from err
 
-            self._mgr = self._mgr.reindex_axis(
-                value.index.copy(), axis=1, fill_value=np.nan
-            )
+            # GH31368 preserve name of index
+            index_copy = value.index.copy()
```
Comment (jreback) [inline pandas/core/frame.py:3210]: FYI `value.index.set_name(self.index.name)` should work as well

File: pandas/tests/indexing/test_partial.py [hunk 2]
```diff
@@ -660,3 +660,12 @@ def test_indexing_timeseries_regression(self):
...
         tm.assert_series_equal(result, expected)
+
+    def test_index_name_empty(self):
+        # GH 31368
+        df = pd.DataFrame({}, index=pd.RangeIndex(0, name="df_index"))
+        series = pd.Series(1.23, index=pd.RangeIndex(4, name="series_index"))
+
+        df["series"] = series
+
+        tm.assert_equal(df.index.name, "df_index")
```
Comment (jreback) [inline pandas/tests/indexing/test_partial.py:671]: can you construct the expected frame and use tm.assert_frame_equal

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
also pls merge master

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@simonjayhawkins ok for 1.1.2 (if you haven't tagged), otherwise let's push to 1.1.3

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins ok for 1.1.2 (if you haven't tagged), otherwise let's push to 1.1.3

ok for 1.1.2

**Comment by simonjayhawkins** [comment]
Thanks @Dr-Irv

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**